pkg/actions/karpenter/create.go

```diff
@@ -27,18 +27,21 @@ func (i *Installer) Create() error {
 			return i.ClientSet, nil
 		},
 	}
-	karpenterServiceAccountTaskTree := i.StackManager.NewTasksToCreateIAMServiceAccounts([]*api.ClusterIAMServiceAccount{
-		{
-			ClusterIAMMeta: api.ClusterIAMMeta{
-				Name:      karpenter.DefaultServiceAccountName,
-				Namespace: karpenter.DefaultNamespace,
+	i.Config.SetDefaultKarpenterCreateServiceAccount()
+	if api.IsDisabled(i.Config.Karpenter.CreateServiceAccount) {
+		karpenterServiceAccountTaskTree := i.StackManager.NewTasksToCreateIAMServiceAccounts([]*api.ClusterIAMServiceAccount{
+			{
+				ClusterIAMMeta: api.ClusterIAMMeta{
+					Name:      karpenter.DefaultServiceAccountName,
+					Namespace: karpenter.DefaultNamespace,
+				},
+				AttachRoleARN: roleArn,
```

Comment:
wait so if `IsDisabled`, we create it? I'm confused :joy:

Reply:
Yes. `createServiceAccount` is set to false, i.e.: Disabled, we create an account. If it's true, we DON'T want to create an account because Karpenter will create it. :)

Reply:
ohh I see. Its not a "do you want eksctl to creates service account", its "do you want to create service account? otherwise eksctl will"